lib/client/db/dbcmd/dbcmd.go

```diff
@@ -515,32 +541,103 @@ func (c *CLICommandBuilder) getSnowflakeCommand() *exec.Cmd {
+func pythonKeywordArgs(args map[string]string) string {
+	var els []string
+
+	for key, value := range args {
+		els = append(els, fmt.Sprintf(",%v=%v", key, value))
 	}
 
-	if c.tc.InsecureSkipVerify {
-		args = append(args, "--insecure")
+	return strings.Join(els, " ")
+}
+
+// getElasticsearchCommand returns a command to connect to Elasticsearch.
+// - we default to `elasticsearch-sql-cli` if we can find it;
+// - a first fallback option is Python with `elasticsearch` package available;
+// - the last option is a `curl` command.
+func (c *CLICommandBuilder) getElasticsearchCommand() *exec.Cmd {
+	if c.isElasticsearchSqlBinAvailable() {
+		if c.options.noTLS {
+			return c.options.exe.Command(elasticsearchSqlBin, fmt.Sprintf("http://%v:%v/", c.host, c.port))
+		} else {
+			c.options.log.Warnf("%v is only supported in --tunnel mode, moving to next client.", elasticsearchSqlBin)
+		}
 	}
 
-	if c.options.caPath != "" {
-		args = append(args, []string{"--cacert", c.options.caPath}...)
+	var curlCommand *exec.Cmd
+
+	if c.options.noTLS {
+		curlCommand = c.options.exe.Command(curlBin, fmt.Sprintf("http://%v:%v/", c.host, c.port))
+	} else {
+		args := []string{
+			fmt.Sprintf("https://%v:%v/", c.host, c.port),
+			"--key", c.profile.KeyPath(),
+			"--cert", c.profile.DatabaseCertPathForCluster(c.tc.SiteName, c.db.ServiceName),
+		}
+
+		if c.tc.InsecureSkipVerify {
+			args = append(args, "--insecure")
+		}
+
+		if c.options.caPath != "" {
+			args = append(args, []string{"--cacert", c.options.caPath}...)
+		}
+
+		// Force HTTP 1.1 when connecting to remote web proxy. Otherwise HTTP2 can
+		// be negotiated which breaks the engine.
+		if c.options.localProxyHost == "" {
+			args = append(args, "--http1.1")
+		}
+
+		curlCommand = c.options.exe.Command(curlBin, args...)
 	}
 
-	// Force HTTP 1.1 when connecting to remote web proxy. Otherwise HTTP2 can
-	// be negotiated which breaks the engine.
-	if c.options.localProxyHost == "" {
-		args = append(args, "--http1.1")
+	if pythonBin, found := c.findPythonBin(); found {
+		if c.checkPythonPackage(pythonBin, "elasticsearch") {
+			kwArgs := map[string]string{}
+
+			var esHost string
+			if c.options.noTLS {
+				esHost = fmt.Sprintf("http://%v:%v/", c.host, c.port)
+			} else {
+				esHost = fmt.Sprintf("https://%v:%v/", c.host, c.port)
+				kwArgs["client_key"] = pythonStringLiteral(c.profile.KeyPath())
+				kwArgs["client_cert"] = pythonStringLiteral(c.profile.DatabaseCertPathForCluster(c.tc.SiteName, c.db.ServiceName))
+
+				if c.options.caPath != "" {
+					kwArgs["ca_certs"] = pythonStringLiteral(c.options.caPath)
+				}
+				if c.tc.InsecureSkipVerify {
+					kwArgs["verify_certs"] = "False"
+				}
+			}
+
+			esPart := fmt.Sprintf("es = Elasticsearch(%v%v)", pythonStringLiteral(esHost), pythonKeywordArgs(kwArgs))
+			script := `
+from elasticsearch import Elasticsearch
+print("# To connect with curl:", %v)
+%v
+print("es =", es)
+print("es.search(): ", es.search())
+`
```

Comment:
I agree this could be better, I was struggling a bit with how to best present it. I had a chat with @smallinsky today exactly about this, and I have an update to this PR in the works that hopefully addresses this. I'll update the PR with "UX" section to elaborate on this; stay tuned!